### PR TITLE
Adding Deprecation links

### PR DESCRIPTION
**Pull request by jamiehannaford**
Note: this PR depends on #555 being merged first. We also need to setup `docs.php-opencloud.com` CNAME (or whatever the subdomain will be) first. I've just noticed that I've referenced `doc.*` - so I'll need to do a search and replace and rebase

### CODE REVIEW

File: docs/userguide/Clients.md [hunk 1]
```diff
@@ -1,149 +1,5 @@
...
-which will result in:
-
-> User-Agent: MyPrefix OpenCloud/xxx cURL/yyy PHP/zzz
-
-where `$client` is an instance of `OpenCloud\OpenStack` or `OpenCloud\Rackspace`.
-
-## 5. Other functionality
-
-For a full list of functionality provided by Guzzle, please consult the [official documentation](http://docs.guzzlephp.org/en/latest/http-client/client.html).
+https://doc.php-opencloud.com/en/latest/index.html
```
Comment (ycombinator) [inline docs/userguide/Clients.md]: Any reason you decided not to have a separate document on Clients in the new documentation?

Comment (jamiehannaford) [inline docs/userguide/Clients.md]: I thought about it, but every service now in its introduction references how to set up the relevant client and service (for both OS and RS). It provides the sample and explains all the arguments in detail, so I didn't think a separate document (that wasn't tied to a specific service) would have much use. What do you think?

Comment (ycombinator) [inline docs/userguide/Clients.md]: I agree that a separate section on setting up OS/RS clients is not useful any more. There are, however, other parts of this document - specifically the parts about User Agents and Loggers - that are not included anywhere else in the new documentation. Do you think we ought to create a separate document (or documents) for them, if we are going to remove this document on Clients?

Comment (jamiehannaford) [inline docs/userguide/Clients.md]: yes, definitely - that was an oversight on my part. I think we should add add them as separate documents and link to them in the "usage tips" toctree in index.rst

Comment (ycombinator) [inline docs/userguide/Clients.md]: Sounds good. Do you want to include them in this PR (I would prefer this as the Clients document is being deleted in this PR) or in a separate PR?

Comment (jamiehannaford) [inline docs/userguide/Clients.md]: yup, I'll add now and rebase too

**Comment by jamiehannaford** [comment]
@ycombinator ok, I've added some more docs - does that cover everything? We need to set up the CNAME _before_ this PR is merged

**Comment by jamiehannaford** [comment]
@ycombinator The DNS migration is done :metal: This is ready for review again

**Pull request merged**